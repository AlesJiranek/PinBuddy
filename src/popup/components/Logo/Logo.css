.logo {
  display: flex;
}

.logo__title {
  margin-right: calc(var(--lh) / 4);
}

.logo__link {

}
