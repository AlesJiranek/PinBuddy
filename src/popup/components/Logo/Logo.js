import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Title, Link } from 'theme';
import './Logo.css';

const Logo = props => {
  const { username, online } = props;
  const shouldDisplayUsername = username && online;
  return (
    <div className="logo">

      <div className="logo__title">
        <Title t={chrome.i18n.getMessage('extensionName')} />
      </div>

      {shouldDisplayUsername && (
        <>
          (
          <Link
            t={username}
            url={`https://pinboard.in/u:${username}`}
            title={chrome.i18n.getMessage('popupOpenYourPinboardProfile')}
            blank
          />
          )
        </>
      )}

    </div>
  );
};

Logo.propTypes = {
  username: PropTypes.string.isRequired,
  online: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  username: state.user.username,
  online: state.online,
});

export default connect(mapStateToProps)(Logo);

