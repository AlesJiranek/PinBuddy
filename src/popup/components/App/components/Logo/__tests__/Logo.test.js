import React from 'react';
import { shallow } from 'enzyme';

import { Logo } from '../Logo';

describe('<Logo />', () => {

  const defaultProps = {
    username: 'pinbuddy',
    online: true,
  };

  it('should render correctly', () => {
    const tree = shallow(
      <Logo
        {...defaultProps}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('should not render when username is falsy', () => {
    const tree = shallow(
      <Logo
        {...defaultProps}
        username=""
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('should not render when online is falsy', () => {
    const tree = shallow(
      <Logo
        {...defaultProps}
        online={false}
      />
    );
    expect(tree).toMatchSnapshot();
  });

});
