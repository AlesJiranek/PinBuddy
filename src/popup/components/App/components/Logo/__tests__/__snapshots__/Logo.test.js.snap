// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should not render when online is falsy 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Logo
    online={false}
    username="pinbuddy"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo__title"
        >
          <Title
            t="extensionName"
          />
        </div>,
        false,
      ],
      "className": "logo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Title
            t="extensionName"
          />,
          "className": "logo__title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "t": "extensionName",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo__title"
          >
            <Title
              t="extensionName"
            />
          </div>,
          false,
        ],
        "className": "logo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Title
              t="extensionName"
            />,
            "className": "logo__title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "t": "extensionName",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Logo /> should not render when username is falsy 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Logo
    online={true}
    username=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo__title"
        >
          <Title
            t="extensionName"
          />
        </div>,
        "",
      ],
      "className": "logo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Title
            t="extensionName"
          />,
          "className": "logo__title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "t": "extensionName",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo__title"
          >
            <Title
              t="extensionName"
            />
          </div>,
          "",
        ],
        "className": "logo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Title
              t="extensionName"
            />,
            "className": "logo__title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "t": "extensionName",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Logo /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Logo
    online={true}
    username="pinbuddy"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo__title"
        >
          <Title
            t="extensionName"
          />
        </div>,
        <React.Fragment>
          (
          <Link
            blank={true}
            t="pinbuddy"
            title="popupOpenYourPinboardProfile"
            url="https://pinboard.in/u:pinbuddy"
          />
          )
        </React.Fragment>,
      ],
      "className": "logo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Title
            t="extensionName"
          />,
          "className": "logo__title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "t": "extensionName",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "(",
            <Link
              blank={true}
              t="pinbuddy"
              title="popupOpenYourPinboardProfile"
              url="https://pinboard.in/u:pinbuddy"
            />,
            ")",
          ],
        },
        "ref": null,
        "rendered": Array [
          "(",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "blank": true,
              "t": "pinbuddy",
              "title": "popupOpenYourPinboardProfile",
              "url": "https://pinboard.in/u:pinbuddy",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          ")",
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo__title"
          >
            <Title
              t="extensionName"
            />
          </div>,
          <React.Fragment>
            (
            <Link
              blank={true}
              t="pinbuddy"
              title="popupOpenYourPinboardProfile"
              url="https://pinboard.in/u:pinbuddy"
            />
            )
          </React.Fragment>,
        ],
        "className": "logo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Title
              t="extensionName"
            />,
            "className": "logo__title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "t": "extensionName",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "(",
              <Link
                blank={true}
                t="pinbuddy"
                title="popupOpenYourPinboardProfile"
                url="https://pinboard.in/u:pinbuddy"
              />,
              ")",
            ],
          },
          "ref": null,
          "rendered": Array [
            "(",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "blank": true,
                "t": "pinbuddy",
                "title": "popupOpenYourPinboardProfile",
                "url": "https://pinboard.in/u:pinbuddy",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            ")",
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
